### DISCUSSION TITLE
Label spacing

### DISCUSSION THREAD

**Discussion by matheusPereiraG** [patrykandpatrick/vico#463] [Questions] [answered]
Hello, i have a question regarding the spacing between x label values. I have this graph and i want to reduce the length between the bars, so they can almost be "touching" each other. I already tried working with the axisLabelComponent in the bottomAxis and then setting the horizontal margin/padding to zero but i didn't saw any changes (as mentioned in https://github.com/patrykandpatrick/vico/discussions/441 )

Also, what are the components that restrict the distance between labels?

Any help is much appreciated and thank you for the amazing library :) 

<img width="327" alt="image" src="https://github.com/patrykandpatrick/vico/assets/32481629/b1014ac2-5241-4a59-b373-019957b01ad8">

**Comment by patrickmichalik** [comment, marked as answer]
Hello! Thank you for the kind words. The spacing between neighboring major entries is specified by `HorizontalDimensions#xSpacing`. Each `Chart` requests a value, and the largest of these values is used. For `ColumnChart`s, the requested value depends on `spacingDp` and `innerSpacingDp`. For `LineChart`s, it depends on `spacingDp` and `LineSpec#pointSizeDp` (unless `LineSpec#point` is `null`). To achieve the desired result, please decrease `ColumnChart#spacingDp` to the desired value and lower `LineChart#spacingDp` such that it’s no greater than the sum of `ColumnChart#spacingDp` and the columns’ width (`LineComponent#thicknessDp`). Regarding the distance between `HorizontalAxis` labels, it depends on `HorizontalDimensions#xSpacing` and the `AxisItemPlacer.Horizontal` implementation used.

**Reply by matheusPereiraG** [reply]
Thank you! The problem was that i had a different spacing value for lineChart and columnChart, setting the spacing to both fixed the issue 🚀

**Reply by patrickmichalik** [reply]
You’re welcome!
